### PR TITLE
Request and approve certs

### PR DESCRIPTION
**Pull request by Kuzirashi**
- Add requesting and approving logic to CertificateSpecificContract
- Make EnergyCertificateBundle and Certificate logic inherit from CertificateSpecificContract
- New Certificate method `requestCertificates(uint _assetId, uint lastRequestedSMReadIndex)` can be called by `AssetManager` role
- New Certificate method `approveCertificationRequest(uint _certicationRequestIndex)` can be called by `Issuer` role

### CODE REVIEW

File: src/test/OriginContractLookup.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ import { CertificateDB } from '../wrappedContracts/CertificateDB';
 import { CertificateLogic } from '../wrappedContracts/CertificateLogic';
 import { OriginContractLookupJSON, CertificateLogicJSON, CertificateDBJSON } from '..';
 
-describe('OriginContractLookup', () => {
+describe.skip('OriginContractLookup', () => {
```
Comment (JosephBagaric) [inline src/test/OriginContractLookup.ts]: Undo skipping?

File: src/wrappedContracts/CertificateLogic.ts [hunk 2]
```diff
@@ -631,12 +631,12 @@ export class CertificateLogic extends GeneralFunctions {
...
     }
 
-    async createTradableEntity(_assetId: number, _powerInW: number, txParams?: SpecialTx) {
+    async requestCertificates(_assetId: number, limitingSmartMeterReadIndex: number, txParams?: SpecialTx) {
         let transactionParams;
 
-        const txData = await this.web3Contract.methods
-            .createTradableEntity(_assetId, _powerInW)
-            .encodeABI();
+        const preparedMethod = this.web3Contract.methods.requestCertificates(_assetId, limitingSmartMeterReadIndex)
```
Comment (JosephBagaric) [inline src/wrappedContracts/CertificateLogic.ts]: Maybe rename `preparedMethod()` to something more descriptive? `preparedMethod` name doesn't give us a hint which method is being prepared by it.

File: contracts/Origin/CertificateLogic.sol [hunk 3]
```diff
@@ -144,12 +143,14 @@ contract CertificateLogic is CertificateInterface, RoleManagement, TradableEntit
         TradableEntityDBInterface(address(db)).setForSale(_certificateId, false);
     }
 
-    /// @notice creates a new Entity / certificate
+    /// @notice creates a new bundle
```
Comment (JosephBagaric) [inline contracts/Origin/CertificateLogic.sol]: Are bundles created in CertificateLogic? Aren't they created in EnergyCertificateBundleLogic?

File: contracts/Origin/CertificateSpecificContract.sol [hunk 4]
```diff
@@ -35,4 +61,86 @@ contract CertificateSpecificContract {
...
+    CertificationRequest[] public certificationRequests;
+
+    function createTradableEntity(
+        uint _assetId,
+        uint _powerInW
+    )
+        internal
+        returns (uint)
+    {
+        return 0;
```
Comment (JosephBagaric) [inline contracts/Origin/CertificateSpecificContract.sol]: Why does this method return 0?

File: contracts/Origin/CertificateLogic.sol (hunk 3 above)
Comment (Kuzirashi) [reply contracts/Origin/CertificateLogic.sol]: It should be `creates a new Entity / certificate`, because it doesn't create bundles

**Pull request merged**